train_controlnet3d.py

```diff
@@ -844,8 +843,10 @@ def load_model_hook(models, input_dir):
         for step, batch in enumerate(train_dataloader):
             with accelerator.accumulate(controlnet):
                 # Convert videos to latent space
-                latents = vae.encode(batch["pixel_values"].to(dtype=weight_dtype)).latent_dist.sample()
-                latents = latents * vae.config.scaling_factor
+                pixel_values = rearrange(batch["pixel_values"], "b f c h w -> (b f) c h w")
+                latents = vae.encode(pixel_values.to(dtype=weight_dtype)).latent_dist.sample()
+                latents *= vae.config.scaling_factor
```

Comment:
Does this work with training? I could be misremembering, but I thought you can't use *= and += if you want gradients.

Reply:
This part doesn't get any gradients as the encoded latents are the target for the loss function. This is why it's fine to cache all of the latent videos as the VAE isn't generally trained anyway.